### PR TITLE
Fix DB transaction retry issue

### PR DESCRIPTION
**Pull request by tobiscr**
See: https://github.com/kyma-incubator/reconciler/issues/755 and https://github.com/kyma-incubator/reconciler/issues/753

### CODE REVIEW

File: pkg/cluster/inventory.go [hunk 1]
```diff
@@ -20,6 +20,7 @@ type Inventory interface {
 	Delete(runtimeID string) error
 	Get(runtimeID string, configVersion int64) (*State, error)
 	GetLatest(runtimeID string) (*State, error)
+	GetAll() ([]*State, error)
```
Comment (tobiscr) [inline pkg/cluster/inventory.go:23]: Required to stabilise inventory unit-tests when using PG

File: pkg/db/interface.go [hunk 2]
```diff
@@ -44,3 +48,15 @@ type DataRows interface {
 	Scan(dest ...interface{}) error
 	Next() bool
 }
+
+func newID() string {
```
Comment (jeremyharisch) [inline pkg/db/interface.go:52]: Why not using uuid package instead? We are already using it in other parts of the project

File: pkg/db/query.go [hunk 3]
```diff
@@ -283,6 +285,24 @@ func (s *Select) GetMany() ([]DatabaseEntity, error) {
 	return result, nil
 }
 
+func (s *Select) sql() string {
```
Comment (jeremyharisch) [inline pkg/db/query.go:288]: It would be nice to have a unit test for this function 👍🏻

Comment (tobiscr) [reply pkg/db/query.go:288]: done

File: pkg/db/interface.go (hunk 2 above)
Comment (tobiscr) [reply pkg/db/interface.go]: Changed to UUID

**Review by jeremyharisch — APPROVED** [review]

**Review by clebs — APPROVED** [review]

**Pull request merged**